Riot/Modules/Application/AppCoordinator.swift

```diff
@@ -49,10 +53,14 @@ final class AppCoordinator: NSObject, AppCoordinatorType {
     
     // MARK: - Setup
     
-    init(router: RootRouterType) {
+    init(router: RootRouterType, window: UIWindow) {
         self.rootRouter = router
         self.customSchemeURLParser = CustomSchemeURLParser()
         self.userSessionsService = UserSessionsService()
+        
+        super.init()
+        
+        setupFlexDebuggerOnWindow(window)
```

Comment:
Good catch for the call to `super.init()`. However the `setupFlexDebuggerOnWindow(window)` it's better placed in the Coordinator `start()`method.

Reply:
I wanted to avoid having to keep a local reference to the window in the AppCoordinator and it doesn't feel right to pass it as a parameter in the start method. `Start` feels more related to the UI stack and what we display on screen than to configuring the application.

Reply:
We should even not give the window here as parameter, anyway it's a detail. Yes the start method is here to make the Coordinator to take control on navigation (instantiate view controllers and start the navigation, https://khanlou.com/2015/10/coordinators-redux/). But will soonly have a flag in `BuildSettings` to configure and enable it on demand, if we can avoid have big init with conditions in it it can be great. We can use a `setup()` method here where all singleton can be instantiated by the AppCoordinator but it's not shocking also to put that in the `start()` for the `AppCoordinator` I mean.